crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1626,21 +1635,21 @@ impl World {
             .map(Into::into)
     }
 
-    /// Registers a new [`Resource`] type and returns the [`ComponentId`] created for it.
+    /// Registers a new non-send resource type and returns the [`ComponentId`] created for it.
     ///
-    /// This enables the dynamic registration of new [`Resource`] definitions at runtime for
+    /// This enables the dynamic registration of new non-send resources definitions at runtime for
     /// advanced use cases.
     ///
     /// # Note
     ///
-    /// Registering a [`Resource`] does not insert it into [`World`]. For insertion, you could use
-    /// [`World::insert_resource_by_id`].
-    pub fn register_resource_with_descriptor(
```

Comment:
Why remove this method?  You left the typed `fn register_resource<R>` method.  

If we want to remove it in favor of calling `register_component_with_descriptor`, I think we should leave it `#[deprecated]` for a release and mention it in the migration guide, and we should also remove `register_resource` for consistency.

Reply:
The reason why I removed `register_resource_with_descriptor` is that it didn't really make sense to have it around anymore, because resources are now very hard-coded components (see macros/src/component.rs). They only exist with one particular `ComponentDescriptor`, so it wouldn't make sense to allow users to register custom resources.

This is the reason why `register_resource<R>()` still exists while `register_resource_with_descriptor` doesn't.